graph/path/yen_ksp.go

```diff
@@ -0,0 +1,144 @@
+// Copyright ©2018 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package path
+
+import (
+	"fmt"
+	"sort"
+
+	"gonum.org/v1/gonum/graph"
+)
+
+type YenShortest struct {
+	path   []graph.Node
+	weight float64
+}
+
+func YenKSP(source graph.Node, sink graph.Node, g graph.Graph, k int) [][]graph.Node {
```

Comment:
This needs documentation; a doc comment above the func for godoc and a comment including a link to the source of the algorithm used here.